tests/test_dataset2.py

```diff
@@ -1,6 +1,12 @@
 import os
+import pytest
+import uuid
 
-from sno.dataset2 import Dataset2, Legend
+from sno.dataset2 import Dataset2, Legend, ColumnSchema, Schema
+
+
+def _uuid():
+    return str(uuid.uuid4())
```

Comment:
Would probably be simpler to have the same UUIDs for different test runs...

```suggestion
@pytest.fixture
def uuid(request):
    """ Deterministic "random" UUID generator seeded from the test ID """
    seed = int(hashlib.sha1(request.node.nodeid).hexdigest(), 16)
    _uuid_gen = random.Random(seed)
    def _uuid():
        return str(uuid.UUID(int=_uuid_gen.getrandbits(128)))
    return _uuid
```

(though should probably move to conftest)

Reply:
Done